src/PHPMailer.php

```diff
@@ -4271,6 +4307,35 @@ public static function isValidHost($host)
         return filter_var('http://' . $host, FILTER_VALIDATE_URL) !== false;
     }
 
+    /**
+     * Check whether the supplied address uses unicode in the localpart.
+     *
+     * @return bool
+     */
+    public function addressHasUnicodeLocalpart($address)
+    {
+        return (bool) preg_match( '/[\x80-\xFF].*@/', $address);
```

Comment:
Is this a reliable pattern? An address can contain multiple @ symbols if they are quoted, so this might match an address like `"user@host@näme"@example.com"` more by luck than design. I can't think of a non-matching address right now though.

Reply:
I believe it to be correct for all valid addresses. `.*` is greedy, so `[\x80-\xff]` will match any non-ASCII byte before the last `@` in the address.

If you feed it garbage (such as a grå@grå.org encoded as cp865 instead of utf8), it'll return true even if the localpart doesn't contain unicode (it contains cp865). GIGO; cp865 isn't valid in email addresses. I don't think the PHP language/runtime gives us a really great option in this case.